### PR TITLE
🐛 Source S3: fix exception when setting CSV stream delimiter to `\t`.

### LINKED ISSUE #32447: Source S3: no longer supports tab-delimited (TSV) configurations
### Connector Name

source-s3

### Connector Version

4.2.0

### What step the error happened?

Configuring a new connector

### Relevant information

The connector no longer honors '\t' (`tab`) into delimiter control of the configuration user interface. This was previously addressed in a workaround pull request #9163. When configuring the connector for TSV files:
1. Documentation insists on use of a two character sequence \t
2. UI Configuration modal accepts a two character sequence \t
3. JSON-serialized configuration states the delimiter is serialized as a string "\\t" 
4. An attempt to test configuration fails with error "Configuration check failed Something went wrong in the connector. See the logs for more details."

[check_connection_source-failure-e1256f04-27d5-4218-9ad2-3b789e07be1e.txt](https://github.com/airbytehq/airbyte/files/13323649/check_connection_source-failure-e1256f04-27d5-4218-9ad2-3b789e07be1e.txt)

5. Integration tests do not feature tab-separated files as use cases

### Relevant log output

```shell
unexpected value; permitted: 'parquet' (type=value_error.const; given=csv; permitted=['parquet'])
streams -> 2 -> format -> filetype
  unexpected value; permitted: 'unstructured' (type=value_error.const; given=csv; permitted=['unstructured'])
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/airbyte/integration_code/main.py", line 42, in <module>
    launch(source, _args)
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 209, in launch
    for message in source_entrypoint.run(parsed_args):
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 109, in run
    yield from map(AirbyteEntrypoint.airbyte_message_to_string, self.check(source_spec, config))
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 133, in check
    check_result = self.source.check(self.logger, config)
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 82, in check
    check_succeeded, error = self.check_connection(logger, config)
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/file_based/file_based_source.py", line 63, in check_connection
    streams = self.streams(config)
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/file_based/file_based_source.py", line 114, in streams
    raise ConfigValidationError(FileBasedSourceError.CONFIG_VALIDATION_ERROR) from exc
airbyte_cdk.sources.file_based.exceptions.ConfigValidationError: Error creating stream config object. Contact Support if you need assistance.
```

### Contribute

- [x] Yes, I want to contribute
```[tasklist]
### Tasks
```

### PR DESCRIPTION
**Pull request by dtiesling**
## What
Closes #32447

Resolves a bug that prevents setting up S3 source streams that are tab-delimited CSVs.

image
image

## How
Updates the pydantic validation of the delimiter to allow `\t`.

## Recommended reading order
1. `csv_format.py`
2. `test_format.py`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*
No changes. Expected behavior should start occurring.

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

Should be a point release since this is a bug fix only.

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

No breaking changes.

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

Unit tests are passing. 
image

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/resources/pull-requests-handbook)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:generateScaffolds` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

<details><summary><strong>Updating the Python CDK</strong></summary>

### Airbyter

Before merging:
- Pull Request description explains what problem it is solving
- Code change is unit tested
- Build and my-py check pass
- Smoke test the change on at least one affected connector
   - On Github: Run [this workflow](https://github.com/airbytehq/airbyte/actions/workflows/connectors_tests.yml), passing `--use-local-cdk --name=source-<connector>` as options
   - Locally: `airbyte-ci connectors --use-local-cdk --name=source-<connector> test`
- PR is reviewed and approved
      
After merging:
- [Publish the CDK](https://github.com/airbytehq/airbyte/actions/workflows/publish-cdk-command-manually.yml)
   - The CDK does not follow proper semantic versioning. Choose minor if this the change has significant user impact or is a breaking change. Choose patch otherwise.
   - Write a thoughtful changelog message so we know what was updated.
- Merge the platform PR that was auto-created for updating the Connector Builder's CDK version
   - This step is optional if the change does not affect the connector builder or declarative connectors.

</details>

### CODE REVIEW

**Comment by dtiesling** [comment]
Hi @evantahler sorry to ping you on this but you were so helpful/responsive with the last issue I encountered with airbyte. The bug affecting tab delimited CSVs is preventing me from using airbyte for a data pipeline and it looks like there is a chance it is low-hanging fruit to fix (I'm short on time so I wasn't able to get full ramped up on developing for airbyte).

Do you have a few minutes to glance at this PR and see if it's as easy a fix as it seems?

File: airbyte-integrations/connectors/source-s3/unit_tests/v4/test_format.py
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-s3/unit_tests/v4/test_format.py:1]: Thank you for including a test!

**Review by natikgadzhi — APPROVED** [review]
The code change LGTM!

Thank you for the contribution, @dtiesling! And apologies for the delay on our side.

@marcosmarxm, teach me the ropes of how to get a PR over the finish line, pretty please. 👀

**Comment by dtiesling** [comment]
> The code change LGTM!
> 
> Thank you for the contribution, @dtiesling! And apologies for the delay on our side.
> 
> @marcosmarxm, teach me the ropes of how to get a PR over the finish line, pretty please. 👀

You're welcome! Thanks for responding. Is there anything left to do on my side?

**Comment by dtiesling** [comment]
@marcosmarxm anything I can do to help keep this moving? I get a 403 when trying to see the failed unit tests. If you can send me the log I can try to help out.

**Comment by marcosmarxm** [comment]
```
________________________ test_csv_format_tab_delimited _________________________

    def test_csv_format_tab_delimited():
>       assert CsvFormat(delimiter=r"\t").delimiter == '\\t'

unit_tests/v4/test_format.py:7: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   pydantic.error_wrappers.ValidationError: 1 validation error for CsvFormat
E   delimiter
E     delimiter should only be one character (type=value_error)

pydantic/main.py:341: ValidationError
=============================== warnings summary ===============================
../../../usr/local/lib/python3.9/site-packages/unstructured/nlp/patterns.py:56
  /usr/local/lib/python3.9/site-packages/unstructured/nlp/patterns.py:56: DeprecationWarning: invalid escape sequence \*
    "\*",  # noqa: W605 NOTE(robinson) - skipping qa because we need the escape for the regex

../../../usr/local/lib/python3.9/site-packages/unstructured/nlp/patterns.py:97
  /usr/local/lib/python3.9/site-packages/unstructured/nlp/patterns.py:97: DeprecationWarning: invalid escape sequence \.
    "[0-9]{1,2}\.[0-9]{1,2}\.[0-9]{1,2}\.[0-9]{1,2}",  # noqa: W605 NOTE(harrell)

../../../usr/local/lib/python3.9/site-packages/unstructured/nlp/patterns.py:103
  /usr/local/lib/python3.9/site-packages/unstructured/nlp/patterns.py:103: DeprecationWarning: invalid escape sequence \.
    IP_ADDRESS_********ME_PATTERN = "[a-zA-Z0-9-]*\.[a-zA-Z]*\.[a-zA-Z]*"  # noqa: W605 NOTE(harrell)

../../../usr/local/lib/python3.9/site-packages/unstructured/nlp/patterns.py:107
  /usr/local/lib/python3.9/site-packages/unstructured/nlp/patterns.py:107: DeprecationWarning: invalid escape sequence \.
    MAPI_ID_PATTERN = "[0-9]*\.[0-9]*\.[0-9]*\.[0-9]*;"  # noqa: W605 NOTE(harrell)

../../../usr/local/lib/python3.9/site-packages/unstructured/nlp/patterns.py:117
  /usr/local/lib/python3.9/site-packages/unstructured/nlp/patterns.py:117: DeprecationWarning: invalid escape sequence \.
    EMAIL_ADDRESS_PATTERN = "[a-z0-9\.\-+_]+@[a-z0-9\.\-+_]+\.[a-z]+"  # noqa: W605 NOTE(harrell)

```

**Comment by dtiesling** [comment]
@marcosmarxm I think I see the problem. I believe I put the test in the wrong place. Looks like the unit tests for sources in CI don't pick up changes in the airbyte-cdk? I'm moving the test to the cdk tests since it makes more sense there and I think it will solve the CI failure.

**Comment by dtiesling** [comment]
Version number is updated and tests have hopefully been fixed in CI.  @marcosmarxm or @natikgadzhi if things look ok could you kick off the tests?

**Comment by marcosmarxm** [comment]
@dtiesling running CI tests again.

**Comment by marcosmarxm** [comment]
Wooohooo finally @dtiesling thanks for the contribution!

**Comment by dtiesling** [comment]
@marcosmarxm 🙌  thanks for shepherding this through! I just noticed in the docs that there is a publish step for updates to the Python CDK. Is there anything needed from me to get that published?

**Comment by dtiesling** [comment]
I made the update to the CDK but I’m not sure how those changes are supposed to trickle down to a connector. My guess is that the cdk change is not present in the connector container. Not sure how to resolve that if it’s the problem though. @marcosmarxm @natikgadzhi @evantahler can any of you provide insights on how CDK changes make it to a connector?

**Comment by marcosmarxm** [comment]
@natikgadzhi let's sync this week to resolve this problem. @dtiesling @blarghmatey any updates return here.

**Comment by blarghmatey** [comment]
It looks like the cdk version needs to be bumped and published to PyPI according to https://github.com/airbytehq/airbyte/pull/35246

Once that is done, the source-s3 connector will need to be re-published, possibly after bumping the poetry.lock file to incorporate the updated cdk version https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-s3/poetry.lock#L5

Is there any possibility of that being completed today? I'm currently blocked on setting up a connection that is needed for powering a lot of downstream pipelines due to this bug.

**Comment by blarghmatey** [comment]
@ambirdsall or @artem1205 it looks like you are typically the ones that kick off the workflow to manually publish the Python CDK? Is that something that can be done to unblock this bug-fix?

**Comment by dtiesling** [comment]
@natikgadzhi @marcosmarxm I think I can knock out two bords with one stone 🐦 🐦 🪨 . I bumped into another issue with the S3 connector that needs to be resolved with an update to the CDK. I _think_ getting that one done by bumping and deploying the CDK version first and then building the S3 connector should fix that issue and pick up these changes at the same time. 

PR for the other fix is https://github.com/airbytehq/airbyte/pull/35650/

cc: @blarghmatey

**Comment by blarghmatey** [comment]
@marcosmarxm @ambirdsall @artem1205 is there any chance that we can get the CDK update published and re-build the S3 source connector? This seems like a pretty common issue that folks are going to be running into.

**Comment by marcosmarxm** [comment]
@blarghmatey we're working on to release the new version with the changes.

**Pull request merged**
